### PR TITLE
test(portal): add sign up tests

### PR DESCRIPTION
**Pull request by bmanifold**

### CODE REVIEW

File: elixir/test/portal_web/live/sign_up_test.exs [hunk 1]
```diff
@@ -0,0 +1,437 @@
+defmodule PortalWeb.SignUpTest do
+  use PortalWeb.ConnCase, async: false
+  import Swoosh.TestAssertions
+  import Portal.DataCase, only: [errors_on: 1]
+  import Portal.AccountFixtures
+
+  setup do
+    # Disable billing for all tests to avoid Stripe API calls
+    Portal.Config.put_env_override(Portal.Billing, enabled: false)
```
Comment (jamilbk) [inline elixir/test/portal_web/live/sign_up_test.exs]: If this is test-wide for all tests does `config/test.exs` make more sense? Then maybe the billing tests should just selectively enable billing.

File: elixir/test/portal_web/live/sign_up_test.exs [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+
+      html =
+        lv
+        |> element("form[phx-submit=submit]")
+        |> render_submit(%{registration: %{email: ""}})
+
+      assert html =~ "can&#39;t be blank"
+    end
+
+    test "handles rate limiting error", %{conn: conn} do
```
Comment (jamilbk) [inline elixir/test/portal_web/live/sign_up_test.exs:385]: Thanks for adding this edge case. Think this test file is a good place to assert on Stripe errors too?

See https://firezone-inc.sentry.io/issues/7157656716/?project=4508756715569152&query=is%3Aunresolved&referrer=issue-stream

File: elixir/test/portal_web/live/sign_up_test.exs [hunk 3]
```diff
@@ -0,0 +1,437 @@
+defmodule PortalWeb.SignUpTest do
+  use PortalWeb.ConnCase, async: false
```
Comment (jamilbk) [inline elixir/test/portal_web/live/sign_up_test.exs]: Also we probably want to try to avoid non-parallel tests as much as possible just to keep CI fast.

Comment (bmanifold) [reply elixir/test/portal_web/live/sign_up_test.exs]: I have no idea why Claude is doing this.  I'll definitely keep a closer eye on this one from now on.

File: elixir/test/portal_web/live/sign_up_test.exs (hunk 2 above)
Comment (bmanifold) [reply elixir/test/portal_web/live/sign_up_test.exs:385]: 🤔 Good question.  Actually, I think I want to go refactor the sign up code.  We really should be putting the `Portal.Billing.provision_account()` call in the `Ecto.Multi` call shouldn't we?

File: elixir/test/portal_web/live/sign_up_test.exs (hunk 1 above)
Comment (bmanifold) [reply elixir/test/portal_web/live/sign_up_test.exs]: As I think about this a little more, I think I'm just going to try to keep billing enabled for these tests.  If billing has been enabled by default for all of the tests this whole time, then I'd be nervous that we might introduce a regression if we turn is off by default and only turn it on for specific billing tests.  Wdyt?

File: elixir/test/portal_web/live/sign_up_test.exs (hunk 2 above)
Comment (jamilbk) [reply elixir/test/portal_web/live/sign_up_test.exs:385]: Yeah probably so that account creation is an all or nothing operation. Though this will be removed in https://github.com/firezone/firezone/issues/8668

File: elixir/test/portal_web/live/sign_up_test.exs (hunk 3 above)
Comment (jamilbk) [reply elixir/test/portal_web/live/sign_up_test.exs]: Yeah it does it for me too. I think when it tries to use `Application.put_env` in the test file itself, it tends to serialize the tests because that's global state.

File: elixir/test/portal_web/live/sign_up_test.exs (hunk 2 above)
Comment (bmanifold) [reply elixir/test/portal_web/live/sign_up_test.exs:385]: How far out are we from actually getting to #8668

Comment (jamilbk) [reply elixir/test/portal_web/live/sign_up_test.exs:385]: Realistically, probably at least a month or two

Comment (bmanifold) [reply elixir/test/portal_web/live/sign_up_test.exs:385]: As we discussed offline, the real fix for this is #8668 and an intermediate fix will probably get messy, so I've added a test that verifies we attempt a retry if the Stripe API returns something other than success.  Not a perfect solution, but it's a stop gap until we can get to the real solution.

File: elixir/test/portal_web/live/sign_up_test.exs [hunk 4]
```diff
@@ -0,0 +1,587 @@
+defmodule PortalWeb.SignUpTest do
+  use PortalWeb.ConnCase, async: true
+  import Swoosh.TestAssertions
+  import Portal.DataCase, only: [errors_on: 1]
+  import Portal.AccountFixtures
+  alias Portal.Billing.Stripe.APIClient
+
+  setup do
```
Comment (jamilbk) [inline elixir/test/portal_web/live/sign_up_test.exs]: Can this be removed?

File: elixir/test/portal_web/live/sign_up_test.exs [hunk 5]
```diff
@@ -0,0 +1,587 @@
...
+    :ok
+  end
+
+  describe "mount/3" do
+    test "initializes socket with sign_up enabled", %{conn: conn} do
```
Comment (jamilbk) [inline elixir/test/portal_web/live/sign_up_test.exs]: Contrary to this test description, these tests seem to make the same assertions with the ones in the `render/1` tests below - perhaps we just need to assert that `mount/3` sets the proper socket state and we can save the content assertions for the `render/1` tests below?

File: elixir/test/portal_web/live/sign_up_test.exs [hunk 6]
```diff
@@ -0,0 +1,587 @@
...
+      assert_email_sent(to: attrs.email)
+
+      # Verify success page is shown
+      html = render(lv)
+      assert html =~ "Your account has been created!"
+      assert html =~ account.name
+      assert html =~ account.slug
+    end
+
+    test "creates default site and internet resource", %{conn: conn} do
```
Comment (jamilbk) [inline elixir/test/portal_web/live/sign_up_test.exs:254]: Definitely good we added these.

File: elixir/test/portal_web/live/sign_up_test.exs [hunk 7]
```diff
@@ -0,0 +1,587 @@
...
+      assert html =~ "Account Name:"
+      assert html =~ "Account Slug:"
+      assert html =~ "Sign In URL:"
+      assert html =~ "Sign In"
+    end
+  end
+
+  describe "Stripe API resilience" do
+    setup do
+      # Enable retry for these tests
```
Comment (jamilbk) [inline elixir/test/portal_web/live/sign_up_test.exs:510]: 👍

**Review by jamilbk — APPROVED** [review]
Found a couple small things otherwise it's good to go.

File: elixir/test/portal_web/live/sign_up_test.exs (hunk 4 above)
Comment (bmanifold) [reply elixir/test/portal_web/live/sign_up_test.exs]: 🤦‍♂️ yes, I can't believe I forgot to remove this.

File: elixir/test/portal_web/live/sign_up_test.exs (hunk 5 above)
Comment (bmanifold) [reply elixir/test/portal_web/live/sign_up_test.exs]: Yeah, that's fair

File: elixir/test/portal_web/live/sign_up_test.exs (hunk 4 above)
Comment (bmanifold) [reply elixir/test/portal_web/live/sign_up_test.exs]: updated.

File: elixir/test/portal_web/live/sign_up_test.exs (hunk 5 above)
Comment (bmanifold) [reply elixir/test/portal_web/live/sign_up_test.exs]: updated

**Review by jamilbk — APPROVED** [review]
